src/decode/rangecoder.rs

```diff
@@ -92,6 +134,23 @@ where
         }
     }
 
+    #[inline]
+    pub fn decode_bit_check(&mut self, prob: u16) -> io::Result<bool> {
```

Comment:
The point of this function is to avoid updating prob (i.e. it's not declared as a `&mut prob`). Otherwise, it does not much differently than the non-`_check` version.

Why not update prob while checking?

Updating probs could cause a scenario where the state is non-recoverable. I think the updated prob value is only meant to be used in the next iteration of the loop anyway, so it should be fine for checking this iteration.

See reference implementation [LzmaDec_TryDummy](https://github.com/OISF/libhtp/blob/0.5.x/htp/lzma/LzmaDec.c#L690).